server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4678,19 +4683,21 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
             copySegmentFiles(storeDirectory, remoteDirectory, null, uploadedSegments, overrideLocal);
 
             if (refreshLevelSegmentSync && remoteSegmentMetadata != null) {
```

Comment:
`org.opensearch.remotestore.RemoteStoreIT.testPeerRecovery*` tests are failing as we pass `refreshLevelSegmentSync=false` for peer recovery. I don't think we need this argument anymore and can always assume refresh level sync.

Reply:
I've updated this to use refresh level sync before peer recovery begins.  Will look at yanking the param entirely in a separate change.